x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/CsvAssert.java

```diff
@@ -369,6 +385,34 @@ private static Comparator<List<Object>> resultRowComparator(List<Type> types) {
         };
     }
 
+    private static Object convertExpectedValue(Type expectedType, Object expectedValue) {
+        if (expectedValue == null) {
+            return null;
+        }
+
+        // convert the long from CSV back to its STRING form
+        return switch (expectedType) {
+            case Type.DATETIME -> rebuildExpected(expectedValue, Long.class, x -> UTC_DATE_TIME_FORMATTER.formatMillis((long) x));
+            case Type.DATE_NANOS -> rebuildExpected(
+                expectedValue,
+                Long.class,
+                x -> DateFormatter.forPattern("strict_date_optional_time_nanos").formatNanos((long) x)
+            );
+            case Type.GEO_POINT, Type.GEO_SHAPE -> rebuildExpected(expectedValue, BytesRef.class, x -> GEO.wkbToWkt((BytesRef) x));
```

Comment:
Should use `UNSPECIFIED.wobToWkt` and merge the Geo and Cartesian into one case statement. In fact we're thinking of moving the code that does not care about Geo vs. Cartesian out of that enum, because it is confusing that we imply that there is a difference, when there is not (in this case).

Reply:
I have moved it as is from above.
If you do not mind, I would like to handle this in a separate PR in case this turns out to be more complex than anticipated (I am new to this area).